clusterloader2/pkg/measurement/common/prometheus_measurement.go

```diff
@@ -56,10 +58,21 @@ type prometheusMeasurement struct {
 }
 
 func (m *prometheusMeasurement) Execute(config *measurement.Config) ([]measurement.Summary, error) {
-	if config.PrometheusFramework == nil {
+	prometheusClient, err := util.GetStringOrDefault(config.Params, "prometheusClient", "inCluster")
+	if err != nil {
+		return nil, err
+	}
+	if prometheusClient != "inCluster" && prometheusClient != "managed" {
+		return nil, fmt.Errorf("unknown Prometheus client")
+	}
+	if prometheusClient == "inCluster" && config.PrometheusFramework == nil {
 		klog.Warningf("%s: Prometheus is disabled, skipping the measurement!", config.Identifier)
 		return nil, nil
 	}
+	if prometheusClient == "managed" && !config.CloudProvider.Features().SupportManagedPrometheus {
+		klog.Warningf("%s: Google Cloud Managed Service for Prometheus is disabled for %q cloud provider, skipping the measurement!", config.CloudProvider, config.Identifier)
```

Comment:
If I configured the measurement in a way that is not supported, I think it should be an error, not warning.

Do we really need this check?
I understand that we want to avoid breaking tests for other cloud providers, but IMHO this should happen on the config level, not here.

Reply:
Removed the warning && added managed Prometheus getter to the cloud provider interface.